app/src/main/java/com/lightricks/feedexercise/network/FeedApiService.kt

```diff
@@ -1,5 +1,39 @@
 package com.lightricks.feedexercise.network
 
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
 /**
  * todo: add the FeedApiService interface and the Retrofit and Moshi code here
  */
+
+
+interface FeedApiService{
+    @GET("Android/demo/feed.json")
+    fun singleFeedItem(): Single<GetFeedResponse>
+
+}
+
+object ServiceBuilder {
+
+    private val BASE_URL = "https://assets.swishvideoapp.com/"
+    private val retrofit = Retrofit.Builder()
+        .baseUrl(BASE_URL)
+        .addConverterFactory(
+            MoshiConverterFactory.create(
+                Moshi.Builder()
+                .add(KotlinJsonAdapterFactory())
+                .add(ThumbnailUrlAdapter())
+                .build()))
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+
+    fun<T> buildService(service: Class<T>): T{
```

Comment:
Instead of a method that builds a **new** service let's make a property that is initialised with the service:
`val service: FeedApiService = retrofit.create(FeedApiService::class.java)`

It means that the service will be created the **first** time somebody will call this property. And the next time the property will be accesses, the existing service will be returned.